pulsar-jms/src/main/java/com/datastax/oss/pulsar/jms/Utils.java

```diff
@@ -457,4 +464,53 @@ static int mapPriorityToPartitionNonLinearly(int priority, int numPartitions) {
     }
     return result;
   }
+
+  /**
+   * Utility to convert path configurations into correct formats for the Apache Pulsar client.
+   * Only values for configuration keys that end with "path", case-insensitive, will be modified.
+   * If the value is a String and begins with "base64:", case-sensitive, it will be base64 decoded then written to a
+   * temp file. The temp file permissions will only allow the current user to access the file. The temp file will be
+   * deleted on JVM exit.
+   * If the value is a String and begins with "classpath:", case-sensitive, the complete filepath will be resolved.
+   * @param configuration - the configuration map to modify in place
+   * @return list of paths representing files created by this method
+   */
+  static List<Path> writeEncodedPathConfigsToTempFiles(Map<String, Object> configuration) {
+    List<Path> createdFiles = new ArrayList<>();
+    for (Map.Entry<String, Object> entry : configuration.entrySet()) {
+      String key = entry.getKey();
+      Object value = entry.getValue();
+      if (key.toLowerCase().endsWith("path") && value instanceof String) {
+        String originalPath = (String) value;
+        String finalPath;
+        if (originalPath.startsWith("base64:")) {
+          try {
+            Path file = Files.createTempFile("pulsar-jms.", ".tmp");
+            file.toFile().deleteOnExit();
+            Files.setPosixFilePermissions(file, PosixFilePermissions.fromString("rw-------"));
+            createdFiles.add(file);
+            String encoded = originalPath.replace("\n", "").replace("\r", "").trim();
+            encoded = encoded.substring("base64:".length());
+            Files.write(file, Base64.getDecoder().decode(encoded));
+            finalPath = file.toAbsolutePath().toString();
+            log.info("Decoded {} to temporary file {}", key, finalPath);
+          } catch (IOException ex) {
+            throw new RuntimeException(
+                    "Cannot decode base64 " + key + " and create temporary file: " + ex, ex);
+          }
+        } else if (originalPath.startsWith("classpath:")) {
+          URL path = Utils.class.getClassLoader().getResource(originalPath.substring("classpath:".length()));
```

Comment:
Using Thread.currentThread().getContextClassload() is generally better, because otherwise you are using only the classloader that loaded the "Utils" class, that is in the pulsar-jms-all jar, and it is not ensured that you are able to access the other resources deployed with the application.

Also usually in JavaEE environments the TCCL (Thread Context Class Loader) is the one you use to load "contextual" resources (Classes with reflection, resource files...)

Reply:
Makes sense, great catch.